tests/integration/conftest.py

```diff
@@ -0,0 +1,128 @@
+import os
+import pytest
+import requests
+from urllib.parse import urlparse
+from requests_kerberos import HTTPKerberosAuth
+
+OPENLCS_URL = os.environ.get('OPENLCS_TEST_URL')
+OPENLCS_LOCAL_LOGIN = os.environ.get('OPENLCS_TEST_LOCAL')
+if not OPENLCS_URL:
+    OPENLCS_URL = 'http://localhost:8000'
+    OPENLCS_LOCAL_LOGIN = True
+OPENLCS_URL = OPENLCS_URL.rstrip('/')
+OPENLCS_LOGIN_DATA = {
+    'username': os.environ.get('OPENLCS_TEST_USER', 'admin'),
+    'password': os.environ.get('OPENLCS_TEST_PASS', 'test'),
+}
+
+
+class OpenLCSTestClient(object):
+    def __init__(self):
+        self.session = requests.Session()
+        self._token = None
+        self.csrf_token = None
+
+    def request(self, endpoint, method='GET', headers=None, auth='default',
+                expected_code=200, **kwargs):
+        headers = dict(headers or {})
+        if self.csrf_token:
+            headers['X-CSRFToken'] = self.csrf_token
+        if auth == 'token':
+            headers['Authorization'] = 'token {}'.format(self.get_token())
+        elif auth == 'default':
+            if OPENLCS_LOCAL_LOGIN and 'sessionid' not in self.session.cookies:
+                # Obtain CSRF token
+                self.request('/', auth=None)
+                self.request(
```

Comment:
Why there need to request ?

Reply:
Via the request to get the CSRF token.

Reply:
I don't understand, why the user/password exist, we still need to get the CSRF token.